Source/SwiftLintCore/Models/Linter.swift

```diff
@@ -209,7 +214,12 @@ public struct CollectedLinter {
 
     private func getStyleViolations(using storage: RuleStorage,
                                     benchmark: Bool = false) -> ([StyleViolation], [(id: String, time: Double)]) {
-        guard !file.isEmpty else {
+        var rules = rules
+        if file.isEmpty {
+            rules = rules.filter(\.shouldLintEmptyFiles)
+        }
```

Comment:
This could be done in the initializer. The file and the rules list exist there already.

Also, this could save one copy by using `rules = file.isEmpty ? linter.rules.filter(\.shouldLintEmptyFiles) : linter.rules`.

Reply:
sure, I can make that change -- I was mostly following the existing logic